### ISSUE TITLE
Unable to enter text inside element declared with id after update to xCode 11 and Appium 1.15.0

### ISSUE THREAD

**Issue by zastress** [appium/appium#13304]
## The problem

After updating to Xcode 11 and Appium 1.15.0 I noticed that I am unable to input text in some fields that are declared by accessibility_id but it works if the element is declared as an xpath on iOS.

## Environment

* Appium version (or git revision) that exhibits the issue: v1.15.0
* Last Appium version that did not exhibit the issue (if applicable): v1.14.2
* Desktop OS/version used to run Appium: MacOS Mojave (10.14.6)
* Node.js version (unless using Appium.app|exe):
* Npm or Yarn package manager: npm 6.11.3
* Mobile platform/version under test: iOS 12, iOS 13
* Real device or emulator/simulator: Real device
* Appium CLI or Appium.app|exe: CLI

## Details

I have a page with multiple text fields and they all have accessibility_id declared as locators.
What is strange is that "Input Text" from AppiumLibrary does not enter the text in the fist one if declared as accessibility_id. I changed the locator to xpath and run the test again.
This time, it entered the text in the first field and also in the second field (that remained declared as id).
Before entering the text inside the field I am executing a find in order to be sure it's there.
In the Appium logs I can see that the element declared with id is correctly identified as being present on the page but as soon as I want to input the text, my test crashes with the error: `WebDriverException: Message: An unknown server-side error occurred while processing the command. Original error: An unknown server-side error occurred while processing the command`

This issue reproduces on both iOS 12 and 13.

I am using Robot Framework and AppiumLibrary (1.5.0.4)
## Link to Appium logs

https://gist.github.com/zastress/f33141acfd94744ed005871987acd029

**Comment by mykola-mokhnach** [comment]
It is known that Apple has changed views representation in the accessibility since Xcode11 (read https://github.com/appium/appium/issues/13288 for more details). Make sure your id matches a proper element (aka text field, currently I see it selects `"value" : "XCUIElementTypeOther",`). Also, I would like to see the actual device log (there might be some useful tracebacks there).

**Comment by zastress** [comment]
So, just found a tweak:
- My element is declared as id (accessibility_id=email_text_field)
- I open de app and go to the element page
- I run a Robot Framework command called (Wait Until Keyword Succeeds) that basically executes a command until a given number of tries is reached without failing the test ( in my case, I run the "Input Text" command for a maximum number of 5 tries). What happens is that first attempt fails and the second one is successful.
Also, at first attempt I can see that the driver executes some kind of long press on the element because I can see the "Paste" popup above the text field.

Here are the Appium logs for this attempt:
https://gist.github.com/zastress/0bae859c87eedc0334a7dc72e4746f30

Here are the phone logs also:
https://gist.github.com/zastress/61182d294451ec709058eceb304128fc

Thank you.

**Comment by mykola-mokhnach** [comment]
Appium verifies if the text input field has the input focus and if the keyboard is opened. If any of these is false then Appium tries to tap the input field, which should eventually make them true. It might be that due to general device slowness or some other problems the keyboard is not shown, so Appium retries the tap in such case. If the overall condition is still false after the second retry then text input fails.

So, as a stable workaround you should first make sure the element you are trying to enter the text into is focused (the `activeElement` selenium API could be used for such purpose). If the element is focused and the keyboard is there then nothing should prevent from typing the text into it.

**Comment by zastress** [comment]
I see. Otherwise, never had this problem. I have been using Appium since early versions and iOS 10. I found a way to tweak it, but in the end, is this a bug or not? Because if I target the same element with xpath is working fine. This could also be because with xpath, locating elements takes longer.
Maybe a solution would be to increase the default timeout in Appium while verifying if the element is focused and the keyboard is present after taping the first time?

**Comment by zastress** [comment]
Just executed my test now with the tweak in place and it failed in the end. On the phone interface I can clearly see that the element is focused and the keyboard is visible. So, in this case, your theory does not apply. Any other ideas?

**Comment by mykola-mokhnach** [comment]
Have you checked the correct element is selected? Does xpath locator return the same element as the id one?

**Comment by zastress** [comment]
Yes, the element used is same as before, same operation, same id. I can see the blinking bar inside this element and also the keyboard is visible. With xpath I target the same element, which is a XCUIElementTypeStaticText. I can confirm that we are talking about the same element.

For now, I will use xpath as a workaround and hope you'll have a clear idea about why this is happening, in the future. Maybe someone else has this problem and will post here.

Thanks.

**Comment by mykola-mokhnach** [comment]
The attached log says the matched element's type is `XCUIElementTypeOther` though. Try more specific predicate locator:

`type == 'XCUIElementTypeTextFiels' AND name == '...'`

**Comment by zastress** [comment]
My bad.... yes, it seems that after update, the accessibility_id is attached to the "TypeOther" element and no longer to the "StaticText". This means that the way Appium "sees" the page is different starting with xCode 11.
Thanks.

**Comment by mykola-mokhnach** [comment]
This means that XCTest now generates different accessibility representation - Appium is just a client to it.
